### PR TITLE
nv4x: enable dma protection

### PR DESCRIPTION
**Pull request by Stojek139808**

### CODE REVIEW

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: src/soc/intel/alderlake/romstage/fsp_params.c [hunk 1]
```diff
@@ -294,6 +294,8 @@ static void fill_fspm_vtd_params(FSP_M_CONFIG *m_cfg,
 	m_cfg->VtdIgdEnable = m_cfg->InternalGfx;
 	m_cfg->VtdIpuEnable = m_cfg->SaIpuEnable;
 
+	m_cfg->PreBootDmaMask = CONFIG(ENABLE_EARLY_DMA_PROTECTION);
```
[the remark this answers is not in the document]
Comment (Stojek139808) [reply src/soc/intel/alderlake/romstage/fsp_params.c:297]: Done.

File: src/soc/intel/alderlake/Kconfig [hunk 2]
```diff
@@ -109,6 +109,7 @@ config CPU_SPECIFIC_OPTIONS
 	select SOC_INTEL_COMMON_BLOCK_SMM
 	select SOC_INTEL_COMMON_BLOCK_SMM_IO_TRAP
 	select SOC_INTEL_COMMON_BLOCK_THERMAL_BEHIND_PMC
+	select SOC_INTEL_COMMON_BLOCK_VTD
```
[the remark this answers is not in the document]
Comment (Stojek139808) [reply src/soc/intel/alderlake/Kconfig:112]: Done.

File: src/soc/intel/alderlake/romstage/romstage.c [hunk 3]
```diff
@@ -163,4 +164,7 @@ void mainboard_romstage_entry(void)
 	pmc_set_disb();
 	if (!s3wake)
 		save_dimm_info();
+
+	if (CONFIG(ENABLE_EARLY_DMA_PROTECTION))
```
[the remark this answers is not in the document]
Comment (Stojek139808) [reply src/soc/intel/alderlake/romstage/romstage.c:168]: Done.

File: configs/config.novacustom_nv4x_tgl_full [hunk 4]
```diff
@@ -33,4 +33,5 @@ CONFIG_EDK2_SD_MMC_TIMEOUT=10
 CONFIG_EDK2_SECURE_BOOT=y
 CONFIG_EDK2_BOOT_MENU_KEY=0x0011
 CONFIG_EDK2_SETUP_MENU_KEY=0x000C
+CONFIG_DRIVERS_EFI_VARIABLE_STORE=y
```
Comment (miczyg1) [inline configs/config.novacustom_nv4x_tgl_full:36]: Full build does not enable DMA protection? I think it should. Same for adl

**Review by miczyg1 — CHANGES_REQUESTED** [review]

**Comment by miczyg1** [comment]
@Stojek139808 I think you need to rebase because conflicts on config files appeared.

**Review by miczyg1 — APPROVED** [review]

**Review by miczyg1 — APPROVED** [review]

**Pull request merged**